packages/protocol/contracts/common/FeeBurner.sol

```diff
@@ -0,0 +1,425 @@
+
+    for (uint256 i = 0; i < tokens.length; i++) {
+      _setDailyBurnLimit(tokens[i], newLimits[i]);
+      _setMaxSplipagge(tokens[i], newMaxSlippages[i]);
+      // Mento tokens don't need to set a router
+      if (newRouters[i] != address(0)) {
+        _setRouter(tokens[i], newRouters[i]);
+      }
+    }
+  }
+
+  // TODO figureout
+  // function setRegistry(address registryAddress) public onlyOwner {
+  //   require(registryAddress != address(0), "Cannot register the null address");
+  //   registry = IRegistry(registryAddress);
+  //   emit RegistrySet(registryAddress);
+  // }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 0, 0, 0);
+  }
+
+  /**
+    * @notice Allows owner to set max slippage for a token.
+    * @param token Address of the token to set.
+    * @param newMax New sllipage to set, as Fixidity fraction.
+    */
+  function setMaxSplipagge(address token, uint256 newMax) external onlyOwner {
+    _setMaxSplipagge(token, newMax);
+  }
+
+  function _setMaxSplipagge(address token, uint256 newMax) private {
+    maxSlippage[token] = FixidityLib.wrap(newMax);
+    emit MAxSlippageSet(token, newMax);
+  }
+
+  /**
+    * @notice Allows owner to set the daily burn limit for a token.
+    * @param token Address of the token to set.
+    * @param newLimit The new limit to set, in the token units.
+    */
+  function setDailyBurnLimit(address token, uint256 newLimit) external onlyOwner {
+    _setDailyBurnLimit(token, newLimit);
+  }
+
+  function _setDailyBurnLimit(address token, uint256 newLimit) private {
+    dailyBurnLimit[token] = newLimit;
+    emit DailyLimitSet(token, newLimit);
+  }
+
+  /**
+    * @notice Allows owner to set the router for a token.
+    * @param token Address of the token to set.
+    * @param router The new router.
+    */
+  function setRouter(address token, address router) external onlyOwner {
+    _setRouter(token, router);
+  }
+
+  function _setRouter(address token, address router) private {
+    routerAddresses[token].push(router);
+    emit RouterAddressSet(token, router);
+  }
+
+  /**
+    * @notice Allows owner to remove a router for a token.
+    * @param token Address of the token.
+    * @param router address of the router to remove.
+    * @param index the index of the router to remove.
+    */
+  function removeRouter(address token, address router, uint256 index) external onlyOwner {
+    // TODO test me
+    require(routerAddresses[token][index] == router, "Index does not match");
+
+    uint256 lenght = routerAddresses[token].length;
+    routerAddresses[token][lenght - 1] = routerAddresses[token][index];
```

Comment:
We also need to move address from length - 1 to removed index

Something like:
address originalAddress = routerAddresses[token][length - 1];
routerAddresses[token][length - 1] = routerAddresses[token][index];
routerAddresses[token][index] = originalAddress;